homeassistant/components/system_bridge/__init__.py

```diff
@@ -141,55 +143,113 @@ async def handle_send_command(call):
-        if device_entry is None:
-            _LOGGER.warning("Missing device: %s", device_id)
-            return
+        coordinator: SystemBridgeDataUpdateCoordinator = hass.data[DOMAIN][
+            call.data[CONF_BRIDGE]
+        ]
+        bridge: Bridge = coordinator.bridge
 
         path = call.data[CONF_PATH]
 
-        entry_id = next(
-            entry.entry_id
-            for entry in hass.config_entries.async_entries(DOMAIN)
-            if entry.entry_id in device_entry.config_entries
-        )
-        coordinator: SystemBridgeDataUpdateCoordinator = hass.data[DOMAIN][entry_id]
-        bridge: Bridge = coordinator.bridge
-
         _LOGGER.debug("Open payload: %s", {CONF_PATH: path})
         try:
             await bridge.async_open({CONF_PATH: path})
-            _LOGGER.debug("Sent open request")
         except (BridgeAuthenticationException, *BRIDGE_CONNECTION_ERRORS) as exception:
-            _LOGGER.warning("Error sending. Error was: %s", exception)
+            raise HomeAssistantError("Error sending") from exception
+        _LOGGER.debug("Sent open request")
+
+    async def handle_send_keypress(call):
+        """Handle the send_keypress service call."""
+        coordinator: SystemBridgeDataUpdateCoordinator = hass.data[DOMAIN][
+            call.data[CONF_BRIDGE]
+        ]
+        bridge: Bridge = coordinator.data
+
+        keyboard_payload: KeyboardPayload = {
+            CONF_KEY: call.data[CONF_KEY],
+            CONF_MODIFIERS: shlex.split(call.data.get(CONF_MODIFIERS, "")),
+        }
+
+        _LOGGER.debug("Keypress payload: %s", keyboard_payload)
+        try:
+            await bridge.async_send_keypress(keyboard_payload)
+        except (BridgeAuthenticationException, *BRIDGE_CONNECTION_ERRORS) as exception:
+            raise HomeAssistantError("Error sending") from exception
+        _LOGGER.debug("Sent keypress request")
+
+    async def handle_send_text(call):
+        """Handle the send_keypress service call."""
+        coordinator: SystemBridgeDataUpdateCoordinator = hass.data[DOMAIN][
+            call.data[CONF_BRIDGE]
+        ]
+        bridge: Bridge = coordinator.data
+
+        keyboard_payload: KeyboardPayload = {CONF_TEXT: call.data[CONF_TEXT]}
+
+        _LOGGER.debug("Text payload: %s", keyboard_payload)
+        try:
+            await bridge.async_send_keypress(keyboard_payload)
+        except (BridgeAuthenticationException, *BRIDGE_CONNECTION_ERRORS) as exception:
+            raise HomeAssistantError("Error sending") from exception
+        _LOGGER.debug("Sent text request")
 
     hass.services.async_register(
         DOMAIN,
         SERVICE_SEND_COMMAND,
         handle_send_command,
-        schema=SERVICE_SEND_COMMAND_SCHEMA,
+        schema=vol.Schema(
+            {
+                vol.Required(CONF_BRIDGE): valid_device,
+                vol.Required(CONF_COMMAND): cv.string,
+                vol.Optional(CONF_ARGUMENTS, []): cv.string,
```

Comment:
The default doesn't match the validator.
```suggestion
                vol.Optional(CONF_ARGUMENTS, ""): cv.string,
```